rs/https_outcalls/consensus/src/payload_builder.rs

```diff
@@ -684,7 +684,10 @@ impl IntoMessages<(Vec<ConsensusResponse>, CanisterHttpBatchStats)>
         let divergece_responses = messages
             .divergence_responses
             .iter()
-            .filter_map(divergence_response_into_reject);
+            .filter_map(|divergence| {
+                stats.divergence_responses += 1;
+                divergence_response_into_reject(divergence)
+            });
```

Comment:
I think we want to do something like:
```suggestion
            .filter_map(divergence_response_into_reject)
            .inspect(|_| stats.divergence_responses += 1);
```
`divergence_response_into_reject` might return `None`, in which case we probably don't want to increase the counter

Reply:
Thanks for pointing out that we should not count None.
Regarding the code, I see that the documentation mentions that `inspect` hints at being used for debugging purposes, therefore I tried enhancing `filter_map`'s callback function instead.

Please let me know what you think

Reply:
IMHO `inspect` is okay in this case, but `filter_map` also works for me.

Reply:
From rust doc:
```
It’s more common for inspect() to be used as a debugging tool than to exist in your final code, but applications may find it useful in certain situations when errors need to be logged before being discarded.
```
Isn't logging (I guess counting) the error exactly our use case here?


I think it's fine, to use `inspect`. It's more concise than logging in the `filer_map` clause, but logging in another closure like `filter_map` is fine too. Although, you would need to then make sure to adhere to the signature, i.e. in `filter_map` return and `Option<T>`.